src/components/environment-badge.tsx

```diff
@@ -37,3 +39,32 @@ export function EnvironmentBadge( { type, selected }: EnvironmentBadgeProps ) {
 
 	return <Badge className={ cx( getClassName() ) }>{ labels[ type ] }</Badge>;
 }
+
+export const useSiteEnvDetails = (
```

Comment:
Have you considered refactoring this hook as a component?
It seems strange to me to have a hook just to return a component function.
I think we could achieve the same thing using a component. What do you think?

Reply:
Could that be a helper function? the logic seems relatively small.

```typescript
export const getEnvironmentType = (connectedSite: SyncSite): EnvironmentType => {
	if (connectedSite.isPressable) {
		return (connectedSite.environmentType as EnvironmentType) ?? 'production';
	}
	return connectedSite.isStaging ? 'staging' : 'production';
};
```